src/codegen/codegen_c_visitor.cpp

```diff
@@ -2737,6 +2737,45 @@ void CodegenCVisitor::print_global_variables_for_hoc() {
     printer->add_line("};");
 }
 
+/**
+ * Return registration type for a given BEFORE/AFTER block
+ * /param block A BEFORE/AFTER block being registered
+ *
+ * Depending on a block type i.e. BEFORE or AFTER and also type
+ * of it's associated block i.e. BREAKPOINT, INITIAL, SOLVE and
+ * STEP, the registration type (as an integer) is calculated.
+ * These values are then interpreted by CoreNEURON internally.
```

Comment:
I guess this is OK for now, but in general it would be nicer if we generated more expressive code, e.g.
```c++
BlockType::BEFORE | BlockType::SOLVE
```
instead of `12`, and made sure that that compiled to the right thing on the [Core]NEURON side.

Reply:
I would create an issue about that and merge this PR if that's okay with you

Reply:
Done in: https://github.com/BlueBrain/nmodl/issues/802

Reply:
> I guess this is OK for now, but in general it would be nicer if we generated more expressive code,

good point!  Just to be sure, you meant hard-coded values of 10/20/1,2.... could be replaced with the enum usage like below:
```cpp
+/*
  * Depending on a block type i.e. BEFORE or AFTER and also type
  * of it's associated block i.e. BREAKPOINT, INITIAL, SOLVE and
  * STEP, the registration type (as an integer) is calculated.
  * These values are then interpreted by CoreNEURON internally.
  */
+enum BABlockTypeValues {
+    BREAKPOINT = 1,
+    SOLVE = 2,
+    INITIAL = 3,
+    STEP = 4,
+    BEFORE = 10,
+    AFTER = 20,
+};
+
+/**
+ * Return registration type for a given BEFORE/AFTER block
+ * /param block A BEFORE/AFTER block being registered
+ */
 static size_t get_register_type_for_ba_block(const ast::Block* block) {
     size_t register_type = 0;
     BAType ba_type;
     /// before block have value 10 and after block 20
     if (block->is_before_block()) {
-        register_type = 10;
+        register_type = BABlockTypeValues::BEFORE;
         ba_type =
             dynamic_cast<const ast::BeforeBlock*>(block)->get_bablock()->get_type()->get_value();
     } else {
-        register_type = 20;
+        register_type = BABlockTypeValues::AFTER;
...
```
that's all?

Reply:
No; I meant that instead of NMODL generating code with magic numbers, it (c|sh)ould generate code with names. Then the mapping from names to numbers can happen in one place ([Core]NEURON) instead of being replicated in two places.

Reply:
Ah ok. I didn’t think much about it as changing API with mod2c in the middle is headache.